main.cpp

```diff
@@ -18,17 +18,14 @@ std::unordered_map<int32_t, struct sigaction> oldSignalHandlers;
 
 void fatalSignalHandler(int32_t signalId)
 {
-    void* array[16];
-    const auto size = backtrace(array, 16);
-    logger::errorImmediate("Fatal signal %d, %d stack frames.", "fatalSignalHandler", signalId, size);
-    logger::logStackImmediate(array, size, "fatalSignalHandler");
-    logger::error("Fatal signal %d, %d stack frames.", "fatalSignalHandler", signalId, size);
-
-    logger::flushLog();
-
-    if (signalId == SIGPIPE)
-    {
-        return;
+    if (signalId != SIGPIPE) {
```

Comment:
Instead of doing this can we just disable the stdout logging on SIGPIPE before logging. So we still have the logs and  back traces on log file?